### ISSUE TITLE
ansible-doc only includes `aliases` in certain circumstances

### ISSUE THREAD

**Issue by felixfontein** [ansible/ansible#81148]
### Summary

When running `ansible-doc --metadata-dump`, the ansible.builtin.systemd_service module's `doc` dict has an `aliases` entry containing `systemd`.

When running `ansible-doc --metadata-dump ansible.builtin` or `ansible-doc --json ansible.builtin.systemd_service`, that `aliases` entry is missing.

### Issue Type

Bug Report

### Component Name

ansible-doc

### Ansible Version

```console
latest devel
```

### Configuration

```console
-
```

### OS / Environment

-

### Steps to Reproduce

Run ansible-doc as mentioned above.

### Expected Results

See above.

### Actual Results

```console
`aliases` should always be there for ansible.builtin.systemd_service.
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by felixfontein** [comment]
I don't mind debugging this myself, but I couldn't find the place where `aliases` is actually added. Any hint on where to find that piece of code is appreciated!

**Comment by bcoca** [comment]
I think it was recently removed with the revamp of plugin loader finding

**Comment by felixfontein** [comment]
Hmm, I really wonder which branch/version I was using that was emitting `aliases`. Because I cannot get it to do that, with neither the stable-2.14, stable-2.15, nor devel branch. And for some reason I was able to do that yesterday with one of these branches :confused:

What I now noticed is that if I install ansible-core with pip, the symlinks will be converted to plain files (since the setup.py hack which prevented that was removed), so in such an installation `ansible-doc --metadata-dump` contains both a systemd and a systemd_service module. But if I run it from an editable install, only the systemd_service module is found, and there's no information that systemd ever existed.

I also noticed that if I create a symlink in a collection without adding a meta/runtime.yml entry, I also cannot find any information on its existence in the `ansible-doc --metadata-dump` output.

Does this mean I have to (again) fall back to filesystem scanning to see whether there are symlink redirects? "So far" (at least a few stable versions back) running `ansible-doc --metadata-dump` and looking at lib/ansible/config/ansible_builtin_runtime.yml and meta/runtime.yml in collections was enough.
